spdx/identifier.go

```diff
@@ -14,13 +14,63 @@ import (
 // ElementIDs should NOT contain the mandatory 'SPDXRef-' portion.
 type ElementID string
 
+// Validate verifies that all the required fields are present.
+// Returns an error if the object is invalid.
+func (e ElementID) Validate() error {
+	if e == "" {
+		return fmt.Errorf("invalid ElementID, must not be blank")
+	}
+
+	return nil
+}
+
+func (e ElementID) String() string {
+	return fmt.Sprintf("SPDXRef-%s", string(e))
+}
+
+// FromString parses an SPDX Identifier string into an ElementID.
+// These strings take the form: "SPDXRef-some-identifier"
+func (e *ElementID) FromString(idStr string) error {
+	idFields := strings.SplitN(idStr, "SPDXRef-", 2)
+	switch len(idFields) {
+	case 2:
+		// "SPDXRef-" prefix was present
+		*e = ElementID(idFields[1])
+	case 1:
```

Comment:
I'm not sure I follow this case: if the `SPDXRef-` prefix is not present, I think we'd want to treat it as an invalid ID rather than ignore that the prefix isn't present? Unless there's a use case here I'm misunderstanding.

Reply:
I think it might be useful for this library to accept slightly malformed data and sorta "clean it up" if the user chooses to export it again. In this case, if the user parsed a file and an SPDXID field was missing the `SPDXRef-` prefix, they could export the file again after parsing it and the file would then pass validation (assuming that was the only issue with the file).

If that happens though, we should probably at least issue a warning rather than just charging ahead, or force the user to opt out of validation in order to parse a document that contains problems like this.

I'll give this one some more thought.